pyccel/parser/semantic.py

```diff
@@ -2054,7 +2098,9 @@ def _visit_For(self, expr, **settings):
         if isinstance(iterable, Variable):
             return ForIterator(target, iterable, body)
 
-        return For(target, iterable, body, local_vars=local_vars)
+        for_expr = For(target, iterable, body, local_vars=local_vars)
+        for_expr.nowait_expr = expr.nowait_expr
```

Comment:
Why not make this a constructor argument then you don't need the setter?

Reply:
i use the setter also in the line 893